### PR TITLE
Add test coverage on edit and renewal transient registrations

### PR DESCRIPTION
**Pull request by cintamani**
This adds request test coverage on relevant forms for edit and renew registration.
This will ensure we have at least one place that test the correct pre-fill of forms in those cases.

### CODE REVIEW

File: spec/requests/waste_exemptions_engine/applicant_email_forms_spec.rb [hunk 1]
```diff
@@ -16,5 +16,25 @@ module WasteExemptionsEngine
         ]
       end
     end
+
+    context "when editing an existing registration" do
+      let(:edit_applicant_email_form) { build(:edit_applicant_email_form) }
+
+      it "prefils applicant email information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/applicant_email_forms_spec.rb]:
```suggestion
      it "pre-fills applicant email information" do
```

File: spec/requests/waste_exemptions_engine/applicant_email_forms_spec.rb [hunk 2]
```diff
@@ -16,5 +16,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/applicant-email/#{edit_applicant_email_form.token}"
+
+        expect(response.body).to include(edit_applicant_email_form.applicant_email)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_applicant_email_form) { build(:renew_applicant_email_form) }
+
+      it "prefils applicant email information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/applicant_email_forms_spec.rb]:
```suggestion
      it "pre-fills applicant email information" do
```

File: spec/requests/waste_exemptions_engine/applicant_name_forms_spec.rb [hunk 3]
```diff
@@ -10,5 +10,27 @@ module WasteExemptionsEngine
       let(:form_data) { { first_name: "Joe", last_name: "Bloggs" } }
       let(:invalid_form_data) { [{ first_name: nil, last_name: nil }] }
     end
+
+    context "when editing an existing registration" do
+      let(:edit_applicant_name_form) { build(:edit_applicant_name_form) }
+
+      it "prefils applicant name information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/applicant_name_forms_spec.rb]:
```suggestion
      it "pre-fills applicant name information" do
```

File: spec/requests/waste_exemptions_engine/applicant_name_forms_spec.rb [hunk 4]
```diff
@@ -10,5 +10,27 @@ module WasteExemptionsEngine
...
+
+        expect(response.body).to include(edit_applicant_name_form.first_name)
+        expect(response.body).to include(edit_applicant_name_form.last_name)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_applicant_name_form) { build(:renew_applicant_name_form) }
+
+      it "prefils applicant name information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/applicant_name_forms_spec.rb]:
```suggestion
      it "pre-fills applicant name information" do
```

File: spec/requests/waste_exemptions_engine/applicant_phone_forms_spec.rb [hunk 5]
```diff
@@ -10,5 +10,25 @@ module WasteExemptionsEngine
       let(:form_data) { { phone_number: "01234567890" } }
       let(:invalid_form_data) { [{ phone_number: nil }, { phone_number: "1234" }] }
     end
+
+    context "when editing an existing registration" do
+      let(:edit_applicant_phone_form) { build(:edit_applicant_phone_form) }
+
+      it "prefils applicant phone information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/applicant_phone_forms_spec.rb]:
```suggestion
      it "pre-fills applicant phone information" do
```

File: spec/requests/waste_exemptions_engine/applicant_phone_forms_spec.rb [hunk 6]
```diff
@@ -10,5 +10,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/applicant-phone/#{edit_applicant_phone_form.token}"
+
+        expect(response.body).to include(edit_applicant_phone_form.phone_number)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_applicant_phone_form) { build(:renew_applicant_phone_form) }
+
+      it "prefils applicant phone information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/applicant_phone_forms_spec.rb]:
```suggestion
      it "pre-fills applicant phone information" do
```

File: spec/requests/waste_exemptions_engine/contact_name_forms_spec.rb [hunk 7]
```diff
@@ -10,5 +10,27 @@ module WasteExemptionsEngine
       let(:form_data) { { first_name: "Joe", last_name: "Bloggs" } }
       let(:invalid_form_data) { [{ first_name: nil, last_name: nil }] }
     end
+
+    context "when editing an existing registration" do
+      let(:edit_contact_name_form) { build(:edit_contact_name_form) }
+
+      it "prefils contact name information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_name_forms_spec.rb]:
```suggestion
      it "pre-fills contact name information" do
```

File: spec/requests/waste_exemptions_engine/contact_name_forms_spec.rb [hunk 8]
```diff
@@ -10,5 +10,27 @@ module WasteExemptionsEngine
...
+
+        expect(response.body).to include(edit_contact_name_form.first_name)
+        expect(response.body).to include(edit_contact_name_form.last_name)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_contact_name_form) { build(:renew_contact_name_form) }
+
+      it "prefils contact name information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_name_forms_spec.rb]:
```suggestion
      it "pre-fills contact name information" do
```

File: spec/requests/waste_exemptions_engine/contact_email_forms_spec.rb [hunk 9]
```diff
@@ -16,5 +16,25 @@ module WasteExemptionsEngine
         ]
       end
     end
+
+    context "when editing an existing registration" do
+      let(:edit_contact_email_form) { build(:edit_contact_email_form) }
+
+      it "prefils contact email information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_email_forms_spec.rb]:
```suggestion
      it "pre-fills contact email information" do
```

File: spec/requests/waste_exemptions_engine/contact_email_forms_spec.rb [hunk 10]
```diff
@@ -16,5 +16,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/contact-email/#{edit_contact_email_form.token}"
+
+        expect(response.body).to include(edit_contact_email_form.contact_email)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_contact_email_form) { build(:renew_contact_email_form) }
+
+      it "prefils contact email information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_email_forms_spec.rb]:
```suggestion
      it "pre-fills contact email information" do
```

File: spec/requests/waste_exemptions_engine/contact_phone_forms_spec.rb [hunk 11]
```diff
@@ -10,5 +10,25 @@ module WasteExemptionsEngine
       let(:form_data) { { phone_number: "01234567890" } }
       let(:invalid_form_data) { [{ phone_number: "123" }, { phone_number: nil }] }
     end
+
+    context "when editing an existing registration" do
+      let(:edit_contact_phone_form) { build(:edit_contact_phone_form) }
+
+      it "prefils contact phone information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_phone_forms_spec.rb]:
```suggestion
      it "pre-fills contact phone information" do
```

File: spec/requests/waste_exemptions_engine/contact_phone_forms_spec.rb [hunk 12]
```diff
@@ -10,5 +10,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/contact-phone/#{edit_contact_phone_form.token}"
+
+        expect(response.body).to include(edit_contact_phone_form.phone_number)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_contact_phone_form) { build(:renew_contact_phone_form) }
+
+      it "prefils contact phone information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_phone_forms_spec.rb]:
```suggestion
      it "pre-fills contact phone information" do
```

File: spec/requests/waste_exemptions_engine/contact_position_forms_spec.rb [hunk 13]
```diff
@@ -12,5 +12,25 @@ module WasteExemptionsEngine
       let(:form_data) { { position: "Chief Waste Carrier" } }
       let(:invalid_form_data) { [] }
     end
+
+    context "when editing an existing registration" do
+      let(:edit_contact_position_form) { build(:edit_contact_position_form) }
+
+      it "prefils contact position information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_position_forms_spec.rb]:
```suggestion
      it "pre-fills contact position information" do
```

File: spec/requests/waste_exemptions_engine/contact_position_forms_spec.rb [hunk 14]
```diff
@@ -12,5 +12,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/contact-position/#{edit_contact_position_form.token}"
+
+        expect(response.body).to include(edit_contact_position_form.position)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_contact_position_form) { build(:renew_contact_position_form) }
+
+      it "prefils contact position information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_position_forms_spec.rb]:
```suggestion
      it "pre-fills contact position information" do
```

File: spec/requests/waste_exemptions_engine/contact_postcode_forms_spec.rb [hunk 15]
```diff
@@ -15,5 +15,25 @@ module WasteExemptionsEngine
                      :contact_postcode_form,
                      request_path: "/contact-postcode/skip_to_manual_address",
                      result_path: "/contact-address-manual"
+
+    context "when editing an existing registration" do
+      let(:edit_contact_postcode_form) { build(:edit_contact_postcode_form) }
+
+      it "prefils contact postcode information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_postcode_forms_spec.rb]:
```suggestion
      it "pre-fills contact postcode information" do
```

File: spec/requests/waste_exemptions_engine/contact_postcode_forms_spec.rb [hunk 16]
```diff
@@ -15,5 +15,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/contact-postcode/#{edit_contact_postcode_form.token}"
+
+        expect(response.body).to include(edit_contact_postcode_form.postcode)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_contact_postcode_form) { build(:renew_contact_postcode_form) }
+
+      it "prefils contact postcode information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/contact_postcode_forms_spec.rb]:
```suggestion
      it "pre-fills contact postcode information" do
```

File: spec/requests/waste_exemptions_engine/operator_name_forms_spec.rb [hunk 17]
```diff
@@ -10,5 +10,25 @@ module WasteExemptionsEngine
       let(:form_data) { { operator_name: "Acme Waste Carriers" } }
       let(:invalid_form_data) { [{ operator_name: nil }] }
     end
+
+    context "when editing an existing registration" do
+      let(:edit_operator_name_form) { build(:edit_operator_name_form) }
+
+      it "prefils operator name information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/operator_name_forms_spec.rb]:
```suggestion
      it "pre-fills operator name information" do
```

File: spec/requests/waste_exemptions_engine/operator_name_forms_spec.rb [hunk 18]
```diff
@@ -10,5 +10,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/operator-name/#{edit_operator_name_form.token}"
+
+        expect(response.body).to include(edit_operator_name_form.operator_name)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_operator_name_form) { build(:renew_operator_name_form) }
+
+      it "prefils operator name information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/operator_name_forms_spec.rb]:
```suggestion
      it "pre-fills operator name information" do
```

File: spec/requests/waste_exemptions_engine/operator_postcode_forms_spec.rb [hunk 19]
```diff
@@ -15,5 +15,25 @@ module WasteExemptionsEngine
                      :operator_postcode_form,
                      request_path: "/operator-postcode/skip_to_manual_address",
                      result_path: "/operator-address-manual"
+
+    context "when editing an existing registration" do
+      let(:edit_operator_postcode_form) { build(:edit_operator_postcode_form) }
+
+      it "prefils operator postcode information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/operator_postcode_forms_spec.rb]:
```suggestion
      it "pre-fills operator postcode information" do
```

File: spec/requests/waste_exemptions_engine/operator_postcode_forms_spec.rb [hunk 20]
```diff
@@ -15,5 +15,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/operator-postcode/#{edit_operator_postcode_form.token}"
+
+        expect(response.body).to include(edit_operator_postcode_form.postcode)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_operator_postcode_form) { build(:renew_operator_postcode_form) }
+
+      it "prefils operator postcode information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/operator_postcode_forms_spec.rb]:
```suggestion
      it "pre-fills operator postcode information" do
```

File: spec/requests/waste_exemptions_engine/site_grid_reference_forms_spec.rb [hunk 21]
```diff
@@ -28,5 +28,27 @@ module WasteExemptionsEngine
                      :site_grid_reference_form,
                      request_path: "/site-grid-reference/skip_to_address",
                      result_path: "/site-postcode"
+
+    context "when editing an existing registration" do
+      let(:edit_site_grid_reference_form) { build(:edit_site_grid_reference_form) }
+
+      it "prefils site grid reference information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/site_grid_reference_forms_spec.rb]:
```suggestion
      it "pre-fills site grid reference information" do
```

File: spec/requests/waste_exemptions_engine/site_grid_reference_forms_spec.rb [hunk 22]
```diff
@@ -28,5 +28,27 @@ module WasteExemptionsEngine
...
+
+        expect(response.body).to include(edit_site_grid_reference_form.grid_reference)
+        expect(response.body).to include(edit_site_grid_reference_form.description)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_site_grid_reference_form) { build(:renew_site_grid_reference_form) }
+
+      it "prefils site grid reference information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/site_grid_reference_forms_spec.rb]:
```suggestion
      it "pre-fills site grid reference information" do
```

File: spec/requests/waste_exemptions_engine/site_postcode_forms_spec.rb [hunk 23]
```diff
@@ -15,5 +15,25 @@ module WasteExemptionsEngine
                      :site_postcode_form,
                      request_path: "/site-postcode/skip_to_manual_address",
                      result_path: "/site-address-manual"
+
+    context "when editing an existing registration" do
+      let(:edit_site_postcode_form) { build(:edit_site_postcode_form) }
+
+      it "prefils site grid reference information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/site_postcode_forms_spec.rb]:
```suggestion
      it "pre-fills site postcode information" do
```

File: spec/requests/waste_exemptions_engine/site_postcode_forms_spec.rb [hunk 24]
```diff
@@ -15,5 +15,25 @@ module WasteExemptionsEngine
...
+        get "/waste_exemptions_engine/site-postcode/#{edit_site_postcode_form.token}"
+
+        expect(response.body).to include(edit_site_postcode_form.postcode)
+      end
+    end
+
+    context "when renewing an existing registration" do
+      let(:renew_site_postcode_form) { build(:renew_site_postcode_form) }
+
+      it "prefils site grid reference information" do
```
Comment (irisfaraway) [inline spec/requests/waste_exemptions_engine/site_postcode_forms_spec.rb]:
```suggestion
      it "pre-fills site postcode information" do
```

**Comment by Cruikshanks** [comment]
🤦‍♂ 😳 oops. Missed that.

File: spec/requests/waste_exemptions_engine/applicant_email_forms_spec.rb (hunk 1 above)
Comment (cintamani) [reply spec/requests/waste_exemptions_engine/applicant_email_forms_spec.rb]: Oh!!!! Really??? That's the way you spell that??? Shut!! I really didn't know 😆 🤣

**Pull request merged**